### PR TITLE
Feature/introduce democracy

### PR DESCRIPTION
**Pull request by mrshiposha**
The democracy setup described [here](https://docs.google.com/document/d/1zXWO0h_NyHcf0IEMHdDhvTpUIXX6a3bMDfIFQvJ7YRY/edit?usp=sharing).

If there are any comment regarding the settings -- please, leave the comments inside the file above.

### CODE REVIEW

**Comment by mrshiposha** [comment]
I removed democracy benchmarks since they don't compile (I took them as is from Substrate).
The democracy pallet will sit here until Polkadot 0.9.39 arrives. So it is a temporary situation.

File: runtime/common/config/pallets/governance.rs [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+	type SwapOrigin = RootOrHalfCouncil;
+	type ResetOrigin = EnsureRoot<AccountId>;
+	type PrimeOrigin = EnsureRoot<AccountId>;
+	type MembershipInitialized = TechnicalCommittee;
+	type MembershipChanged = TechnicalCommittee;
+	type MaxMembers = TechnicalMaxMembers;
+	type WeightInfo = pallet_membership::weights::SubstrateWeight<Runtime>;
+}
+
+impl pallet_scheduler::Config for Runtime {
```
Comment (CertainLach) [inline runtime/common/config/pallets/governance.rs:270]: As I remember, one of the discussion's key points was making democracy execute calls on behalf of some other origin, but here I see it still works with a standard scheduler, which runs calls as root. Do I miss something?

**Review by CertainLach — CHANGES_REQUESTED** [review]

File: runtime/common/config/pallets/governance.rs (hunk 1 above)
Comment (mrshiposha) [reply runtime/common/config/pallets/governance.rs:270]: We decided to use the plain scheduler for democracy for safety reasons.

File: pallets/maintenance/src/lib.rs [hunk 2]
```diff
@@ -54,6 +54,9 @@ pub mod pallet {
 		/// The preimage provider with which we look up call hashes to get the call.
 		type Preimages: QueryPreimage + StorePreimage;
 
+		/// The Origin that has the right to enable or disable the maintenance mode
```
[the remark this answers is not in the document]
Comment (mrshiposha) [reply pallets/maintenance/src/lib.rs:57]: @kozyrevdev, maybe it is too much for the technical committee to execute preimages as **root**? Maybe we should preserve the origin?

File: runtime/common/config/pallets/governance.rs (hunk 1 above)
Comment (CertainLach) [reply runtime/common/config/pallets/governance.rs:270]: But how do we prevent a malicious transaction from being executed?

Comment (mrshiposha) [reply runtime/common/config/pallets/governance.rs:270]: This scheduler is meant to be used **solely** by the democracy.

Comment (mrshiposha) [reply runtime/common/config/pallets/governance.rs:270]: @kozyrevdev

Comment (mrshiposha) [reply runtime/common/config/pallets/governance.rs:270]: Summary of a private conversation with Yaroslav: the Fellowship solves this problem.

File: node/cli/src/chain_spec.rs [hunk 3]
```diff
@@ -172,6 +172,28 @@ macro_rules! testnet_genesis {
 			sudo: SudoConfig {
 				key: Some($root_key),
 			},
+
+			#[cfg(not(feature = "quartz-runtime"))]
```
Comment (kozyrevdev) [inline node/cli/src/chain_spec.rs:176]: unique-runtime

File: node/cli/src/chain_spec.rs [hunk 4]
```diff
@@ -172,6 +172,28 @@ macro_rules! testnet_genesis {
...
+			democracy: DemocracyConfig::default(),
+
+			#[cfg(not(feature = "quartz-runtime"))]
+			council: CouncilConfig::default(),
+
+			#[cfg(not(feature = "quartz-runtime"))]
+			fellowship: FellowshipConfig::default(),
```
Comment (kozyrevdev) [inline node/cli/src/chain_spec.rs:183]: take original fellowship pallet

File: pallets/democracy/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+[package]
+name = "pallet-democracy"
```
Comment (kozyrevdev) [inline pallets/democracy/Cargo.toml:2]: выйти из форка, парити приняла наши изменения

File: pallets/maintenance/src/lib.rs [hunk 6]
```diff
@@ -54,6 +54,9 @@ pub mod pallet {
...
+		type ManagerOrigin: EnsureOrigin<<Self as frame_system::Config>::RuntimeOrigin>;
```
Comment (kozyrevdev) [inline pallets/maintenance/src/lib.rs:58]: rename TechnicalCommitteeMemberOrigin

File: pallets/maintenance/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (kozyrevdev) [reply pallets/maintenance/src/lib.rs:57]: преимаджи вставки идентити должны выполняться от лица TechnicalCommitteeMemberOrigin or Root. Паллета Maintenance делает вызов от лица TechnicalCommitteeMemberOrigin

File: pallets/maintenance/src/lib.rs [hunk 7]
```diff
@@ -81,7 +84,7 @@ pub mod pallet {
 		#[pallet::call_index(0)]
 		#[pallet::weight(<T as Config>::WeightInfo::enable())]
 		pub fn enable(origin: OriginFor<T>) -> DispatchResult {
-			ensure_root(origin)?;
+			T::ManagerOrigin::ensure_origin(origin)?;
```
Comment (kozyrevdev) [inline pallets/maintenance/src/lib.rs:87]: rename

File: pallets/maintenance/src/lib.rs [hunk 8]
```diff
@@ -93,7 +96,7 @@ pub mod pallet {
 		#[pallet::call_index(1)]
 		#[pallet::weight(<T as Config>::WeightInfo::disable())]
 		pub fn disable(origin: OriginFor<T>) -> DispatchResult {
-			ensure_root(origin)?;
+			T::ManagerOrigin::ensure_origin(origin)?;
```
Comment (kozyrevdev) [inline pallets/maintenance/src/lib.rs:99]: rename

File: node/cli/src/chain_spec.rs (hunk 3 above)
Comment (kozyrevdev) [reply node/cli/src/chain_spec.rs:176]: switch to governance feature

File: runtime/common/config/pallets/mod.rs [hunk 9]
```diff
@@ -128,6 +131,13 @@ impl pallet_maintenance::Config for Runtime {
 	type RuntimeEvent = RuntimeEvent;
 	type RuntimeOrigin = RuntimeOrigin;
 	type RuntimeCall = RuntimeCall;
+
+	#[cfg(feature = "governance")]
+	type ManagerOrigin = governance::RootOrAllTechnicalCommittee;
+
+	#[cfg(not(feature = "governance"))]
+	type ManagerOrigin = frame_system::EnsureRoot<AccountId>;
```
Comment (kozyrevdev) [inline runtime/common/config/pallets/mod.rs:139]: rename

File: runtime/common/config/xcm/mod.rs [hunk 10]
```diff
@@ -229,7 +231,13 @@ impl cumulus_pallet_xcmp_queue::Config for Runtime {
 	type ChannelInfo = ParachainSystem;
 	type VersionWrapper = PolkadotXcm;
 	type ExecuteOverweightOrigin = frame_system::EnsureRoot<AccountId>;
-	type ControllerOrigin = EnsureRoot<AccountId>;
+
+	#[cfg(feature = "governance")]
+	type ControllerOrigin = governance::RootOrAllTechnicalCommittee;
```
Comment (kozyrevdev) [inline runtime/common/config/xcm/mod.rs:236]: unify naming

File: pallets/maintenance/src/lib.rs (hunk 6 above)
Comment (CertainLach) [reply pallets/maintenance/src/lib.rs:58]: Это не совсем корректно, в рантайме сюда может быть передан произвольный EnsureOrigin
Лучше тут придерживаться абстракции, т.е maintenance может управлять ManagerOrigin, где ManagerOrigin = TechnicalCommittee.

File: runtime/common/construct_runtime.rs [hunk 11]
```diff
@@ -59,6 +59,30 @@ macro_rules! construct_runtime {
...
 				Preimage: pallet_preimage = 41,
 
+                #[cfg(feature = "governance")]
+                Democracy: pallet_democracy = 42,
+
+                #[cfg(feature = "governance")]
+                Council: pallet_collective::<Instance1> = 43,
+
+                #[cfg(feature = "governance")]
+                Fellowship: pallet_collective::<Instance2> = 44,
```
Comment (kozyrevdev) [inline runtime/common/construct_runtime.rs:69]: use true Fellowship pallet

File: pallets/maintenance/src/lib.rs (hunk 6 above)
Comment (CertainLach) [reply pallets/maintenance/src/lib.rs:58]: Иначе можно напрямую обращаться к палете технического комитета, и лишняя прослойка в виде конфига тут не нужна.

**Pull request closed without merging**